website/docs/r/hybrid_kubernetes_connected_cluster.html.markdown

```diff
@@ -0,0 +1,98 @@
+---
+subcategory: "HybridKubernetes"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_hybrid_kubernetes_connected_cluster"
+description: |-
+  Manages a HybridKubernetes connected cluster.
+---
+
+# azurerm_hybrid_kubernetes_connected_cluster
+
+Manages a HybridKubernetes connected cluster.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "test" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_hybrid_kubernetes_connected_cluster" "test" {
+  name                         = "example-connected-cluster"
+  resource_group_name          = azurerm_resource_group.test.name
+  agent_public_key_certificate = "xxxxxxxxxxxxxxxxxxx"
+  distribution                 = "gke"
+
+  identity {
+    type = "SystemAssigned"
+  }
+  infrastructure = "gcp"
+  location       = "West Europe"
+
+  tags = {
+    ENV = "Test"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this HybridKubernetes Connected Cluster. Changing this forces a new HybridKubernetes Connected Cluster to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the HybridKubernetes connected cluster should exist. Changing this forces a new HybridKubernetes connected cluster to be created.
+
+* `agent_public_key_certificate` - (Required) Base64 encoded public certificate used by the agent to do the initial handshake to the backend services in Azure. Changing this forces a new HybridKubernetes connected cluster to be created.
+
+* `identity` - (Required) An `identity` block as defined below.
+
+* `location` - (Required) The Azure Region where the HybridKubernetes connected cluster should exist. Changing this forces a new HybridKubernetes connected cluster to be created.
+
+* `distribution` - (Optional) The HybridKubernetes distribution running on this connected cluster. Changing this forces a new HybridKubernetes connected cluster to be created.
```

Comment:
three things:

1.

```suggestion
* `distribution` - (Optional) Specifies the distribution running on this Arc Kubernetes Cluster. Changing this forces a new Arc Kubernetes Cluster to be created.
```

2. What does this do?
3. Should this be a user-specifiable field? is this computed-only?

Reply:
Both **distribution** and **infrastructure** are user-specifiable fields. **infrastructure** refers to the cloud provider where the Kubernetes cluster resides (like Azure, aws, gcp). **distribution** refers to the type of the Kubernetes cluster itself(like aks, gke, eks). However, these values need to be manually configured by users. Currently, both of these fields are simple strings only and will not be validated by the back end.